src/DockWidget.cpp

```diff
@@ -515,23 +516,36 @@ void CDockWidget::flagAsUnassigned()
 //============================================================================
 bool CDockWidget::event(QEvent *e)
 {
-	if (e->type() == QEvent::WindowTitleChange)
-	{
-		const auto title = windowTitle();
-		if (d->TabWidget)
-		{
-			d->TabWidget->setText(title);
-		}
-		if (d->ToggleViewAction)
-		{
-			d->ToggleViewAction->setText(title);
-		}
-		if (d->DockArea)
-		{
-			d->DockArea->markTitleBarMenuOutdated();//update tabs menu
-		}
-		emit titleChanged(title);
-	}
+    switch (e->type())
+    {
+#ifndef QT_NO_ACTION
+        case QEvent::Hide:
+            emit visibilityChanged(false);
+            break;
+        case QEvent::Show:
+            emit visibilityChanged(geometry().right() >= 0 && geometry().bottom() >= 0);

```

Comment:
Whats the reason for emitting the test here: geometry().right() >= 0 && geometry().bottom() >= 0?

Reply:
Will this make more sense? This is to make sure that the widget is really visible before emitting visibilityChanged event.
Code comes from here: https://code.qt.io/cgit/qt/qtbase.git/tree/src/widgets/widgets/qdockwidget.cpp?h=dev#n1526

```c++
    case QEvent::Show: {
            QPoint parentTopLeft(0, 0);
            if (isWindow()) {
                if (const QWindow *window = windowHandle())
                    parentTopLeft = window->screen()->availableVirtualGeometry().topLeft();
                else
                    parentTopLeft = QGuiApplication::primaryScreen()->availableVirtualGeometry().topLeft();
            }
            emit visibilityChanged(geometry().right() >= parentTopLeft.x() && geometry().bottom() >= parentTopLeft.y());
        }
```